genesyscloud/routing_queue_conditional_group_routing/genesyscloud_routing_queue_conditional_group_routing_proxy.go

```diff
@@ -0,0 +1,172 @@
+}
+
+// newRoutingQueueConditionalGroupRoutingProxy initializes the Routing queue conditional group routing proxy with all of the data needed to communicate with Genesys Cloud
+func newRoutingQueueConditionalGroupRoutingProxy(clientConfig *platformclientv2.Configuration) *routingQueueConditionalGroupRoutingProxy {
+	api := platformclientv2.NewRoutingApiWithConfig(clientConfig)
+	return &routingQueueConditionalGroupRoutingProxy{
+		clientConfig:                           clientConfig,
+		routingApi:                             api,
+		getAllRoutingQueueConditionRoutingAttr: getAllRoutingQueuesFn,
+		getRoutingQueueConditionRoutingAttr:    getRoutingQueueConditionRoutingFn,
+		updateRoutingQueueConditionRoutingAttr: updateRoutingQueueConditionRoutingFn,
+	}
+}
+
+// getRoutingQueueConditionalGroupRoutingProxy retrieves all Genesys Cloud Routing queue conditional group routing
+func getRoutingQueueConditionalGroupRoutingProxy(clientConfig *platformclientv2.Configuration) *routingQueueConditionalGroupRoutingProxy {
+	if internalProxy == nil {
+		internalProxy = newRoutingQueueConditionalGroupRoutingProxy(clientConfig)
+	}
+
+	return internalProxy
+}
+
+// getAllRoutingQueues gets all routing queues in an org
+func (p *routingQueueConditionalGroupRoutingProxy) getAllRoutingQueues(ctx context.Context) (*[]platformclientv2.Queue, *platformclientv2.APIResponse, error) {
+	return p.getAllRoutingQueueConditionRoutingAttr(ctx, p)
+}
+
+// getRoutingQueueConditionRouting gets the conditional group routing rules for a queue
+func (p *routingQueueConditionalGroupRoutingProxy) getRoutingQueueConditionRouting(ctx context.Context, queueId string) (*[]platformclientv2.Conditionalgrouproutingrule, *platformclientv2.APIResponse, error) {
+	return p.getRoutingQueueConditionRoutingAttr(ctx, p, queueId)
+}
+
+// updateRoutingQueueConditionRouting updates the conditional group routing rules for a queue
+func (p *routingQueueConditionalGroupRoutingProxy) updateRoutingQueueConditionRouting(ctx context.Context, queueId string, rules *[]platformclientv2.Conditionalgrouproutingrule) (*[]platformclientv2.Conditionalgrouproutingrule, *platformclientv2.APIResponse, error) {
+	return p.updateRoutingQueueConditionRoutingAttr(ctx, p, queueId, rules)
+}
+
+// getAllRoutingQueuesFn is an implementation function for getting all queues in an org
+func getAllRoutingQueuesFn(ctx context.Context, p *routingQueueConditionalGroupRoutingProxy) (*[]platformclientv2.Queue, *platformclientv2.APIResponse, error) {
+	var allQueues []platformclientv2.Queue
+	const pageSize = 100
+
+	queues, resp, err := p.routingApi.GetRoutingQueues(1, pageSize, "", "", nil, nil, nil, false)
+	if err != nil {
+		return nil, resp, fmt.Errorf("failed to get routing queues: %s", err)
+	}
+
+	if queues.Entities == nil || len(*queues.Entities) == 0 {
+		return &allQueues, resp, nil
+	}
+
+	for _, queue := range *queues.Entities {
+		allQueues = append(allQueues, queue)
+	}
```

Comment:
we might as well start using the one line approach here and below, since the linter will be catching it in the future e.g. `list = append(list, *entities...)`

Reply:
Will do